<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin Jekyll SEO tag v2.8.0 -->
<title>Hypothesis Testing with BPP | Botany 2023 Workshop on Phylogenetic Networks</title>
<meta name="generator" content="Jekyll v3.9.3" />
<meta property="og:title" content="Hypothesis Testing with BPP" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Methodological advances in reticulate evolution" />
<meta property="og:description" content="Methodological advances in reticulate evolution" />
<link rel="canonical" href="http://localhost:4000/Botany2023-Networks/activities/BPP/" />
<meta property="og:url" content="http://localhost:4000/Botany2023-Networks/activities/BPP/" />
<meta property="og:site_name" content="Botany 2023 Workshop on Phylogenetic Networks" />
<meta property="og:type" content="website" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Hypothesis Testing with BPP" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","description":"Methodological advances in reticulate evolution","headline":"Hypothesis Testing with BPP","url":"http://localhost:4000/Botany2023-Networks/activities/BPP/"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/Botany2023-Networks/assets/main.css"><link type="application/atom+xml" rel="alternate" href="http://localhost:4000/Botany2023-Networks/feed.xml" title="Botany 2023 Workshop on Phylogenetic Networks" /></head>
<body><header class="site-header" role="banner">

  <div class="wrapper"><a class="site-title" rel="author" href="/Botany2023-Networks/">Botany 2023 Workshop on Phylogenetic Networks</a><nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.032C17.335,0,18,0.665,18,1.484L18,1.484z M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.032C17.335,6.031,18,6.696,18,7.516L18,7.516z M18,13.516C18,14.335,17.335,15,16.516,15H1.484 C0.665,15,0,14.335,0,13.516l0,0c0-0.82,0.665-1.483,1.484-1.483h15.032C17.335,12.031,18,12.695,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger"><a class="page-link" href="/Botany2023-Networks/about/">About</a><a class="page-link" href="/Botany2023-Networks/people/">People</a><a class="page-link" href="/Botany2023-Networks/schedule/">Schedule</a><a class="page-link" href="/Botany2023-Networks/pre-workshop/">Pre-workshop</a><a class="page-link" href="/Botany2023-Networks/conduct/">Code of Conduct</a></div>
      </nav></div>
</header>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post">

  <header class="post-header">
    <h1 class="post-title">Hypothesis Testing with BPP</h1>
  </header>

  <div class="post-content">
    <h2 id="introduction">Introduction</h2>

<h2 id="what-is-a-good-model">What is a Good Model?</h2>

<h3 id="in-defense-of-the-biologist">In defense of the biologist</h3>

<p>Full-likelihood method provide us a powerful tool because we can evaluate multiple competing hypotheses in a statistically sound probabilistic framework. However, it can be tempting to compare as many competing models as possible. In practice, this will be difficult due to computational burdens or our inability to explain the rational for every <em>possible</em> model. A successful application of BPP or similar full-likelihood programs relies on your knowledge of the system and their natural history, and perhaps prior analyses with a range of phylogenetic or population genetic tools.</p>

<h3 id="how-to-specify-a-network">How to specify a network</h3>

<p>Formatting networks correctly can be difficult, especially when we want to specify the direction of gene flow. Luckily, BPP comes with a simple program to help create network models based on simpler binary trees. Consider our backbone topology - we will add some node labels for our convenience too. We can create a simple text file <code class="language-plaintext highlighter-rouge">msci-create.in</code> that provides the backbone topology as well as the branche that gene flow goes from and the branch that gene flow goes into. Multiple gene flow events can be specified here and the branch lengths of the introgression edges can be constrained if appropriate:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>tree ((Pmun,Pspe)s,((Dexp,Dint)u,(Dgol,(Dcel,Dlud)w)v)t)r;
hybridization v Dgol, w Dcel as i j tau=yes, yes phi=0.5
</code></pre></div></div>
<p>Here, we have allowed the branch lengths of the two introgression edges to vary and specify an initial <em>introgression probability</em> of 0.5. The <em>introgression probability</em> will be sampled by the MCMC algorithm, so this is only a starting value. We can generate the network used for our analyses with the following command:</p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>~/programs/bpp-4.6.1-macos-x86_64/bin/bpp --msci-create msci-create.in &gt; msci-create.out
</code></pre></div></div>
<p>Below is the resulting network that we can add to our control file. You should try to draw the network yourself from the text string and convince yourself this is correct.</p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>((Pmun,Pspe)s, ((Dexp,Dint)u,((j[&amp;phi=0.500000,tau-parent=yes],Dgol)i,((Dcel)j[&amp;phi=0.500000,tau-parent=yes],Dlud)w)v)t)r;
</code></pre></div></div>

<h2 id="marginal-likelihoods-and-model-probabilities">Marginal likelihoods and model probabilities</h2>
<p>Likelihood methods are great because they often allow us to compare and rank models. But a complication of Bayesian methods is that we have a sample of likelihoods rather than a single maximum likelihood estimate. Additionally, Bayesian estimators will be influenced by the prior distributions we imposed. There has been rigorous investigation into methods for estimating the <em>marginal likehood</em> where those confounding effects of the prior are factored out. The stepping-stone method [CITE] has had good performance in a range of other phylogenetic applications and we will use it again here.</p>

<p>Notes on stepping stone</p>

<p>dos Reis package</p>

<p>our results</p>

<h2 id="bayes-factor-approximation-with-the-savage-dickey-density-ratio">Bayes factor approximation with the Savage-Dickey density ratio</h2>

<h2 id="limits-to-biological-interpretations">Limits to Biological Interpretations</h2>
<p>In our <em>Dryopteris</em> test case, we are lucky such that the biology is already worked out over years of meticulous chromosome work combined with multiple molecular studies. Unknown systems do not have this advantage and the lifetime of an investigation may not be long enough for long-term experimental work. It can be tempting to use network methods to infer the process of hybridization or polyploidization, but I strongly caution against this. Processes other than hybrid speciation can explain high introgression probabilities [CITE] and inferring extant parental species can be dubious if there are sampling gaps among species.</p>

  </div>

</article>

      </div>
    </main><footer class="site-footer h-card">
  <data class="u-url" href="/Botany2023-Networks/"></data>

  <div class="wrapper">

    <h2 class="footer-heading">Botany 2023 Workshop on Phylogenetic Networks</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li class="p-name">Botany 2023 Workshop on Phylogenetic Networks</li><li><a class="u-email" href="mailto:g.tiley@kew.org">g.tiley@kew.org</a></li></ul>
      </div>

      <div class="footer-col footer-col-2"><ul class="social-media-list"><li><a href="https://github.com/gtiley"><svg class="svg-icon"><use xlink:href="/Botany2023-Networks/assets/minima-social-icons.svg#github"></use></svg> <span class="username">gtiley</span></a></li></ul>
</div>

      <div class="footer-col footer-col-3">
        <p>Methodological advances in reticulate evolution</p>
      </div>
    </div>

  </div>

</footer>
</body>

</html>
